### DISCUSSION TITLE
Trouble using Plot library in a CommonJS environment

### DISCUSSION THREAD

**Discussion by AlbertIspiryan-dev** [observablehq/plot#2042] [Q&A]
I'm trying to use the Plot library in a Node.js project, but I'm encountering issues due to the library being distributed as an ES module. 
Here's the relevant part of my `tsconfig.json`:

```json
{
  "compilerOptions": {
    "compilerOptions": {
    "rootDir": "./src",
    "outDir": "dist",
    "target": "esnext",
    "module": "commonjs",
    "esModuleInterop": true,
    "sourceMap": true,
    "declaration": true,
    "declarationMap": true,
    "removeComments": true,
    "skipLibCheck": true,
    "noImplicitAny": true,
    "composite": true,
    "noImplicitReturns": true,
    // other options...
  }
}
```
The error is:

```
Error [ERR_REQUIRE_ESM]: require() of ES Module C:\path\node_modules\d3\src\index.js from C:\path\packages\libs\dist\jsfile.js not supported.
```

Usage:
```
import * as Plot from '@observablehq/plot';
```

**Comment by AlbertIspiryan-dev** [comment]
For the context, I want to use the d3+plot packages on server side of node application to generate a SVG chart and combine it with the JSDOM. Then SVG should be placed in my PDF. As a pdf generator package I'm using `pdfmake`. 

If you can suggest any solution or share ideas with me on how all of the packages can work together, any help much appreciated.

**Comment by mbostock** [comment]
Please read this guide on ES modules: https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c

You probably need to add `"type": "module"` to your `package.json` file.
